hs/app/reachpc/ReachPC/CommandLine.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+module ReachPC.CommandLine (CliOptions(..), parseCliOptions, helpMessage) where
+
+import Data.Text (Text)
+import NeatInterpolation (text)
+import Data.List.Extra (splitOn)
+import System.Environment (getArgs)
+
+helpMessage :: Text
+helpMessage = [text|
+reach - Reach command-line tool
+
+Usage: reach [OPTIONS] COMMAND [ARGS...]
+
+Provide a COMMAND and ARGS to be forwarded to Reach Cloud or Reach Local,
+depending on your configuration. If COMMAND is one of the following special
+commands, its behavior will be executed by this program instead.
+
+Run `reach help` to see the remote's help message.
+
+Available options:
+  -h,--help         Show this help message
+  -e,--env LIST     Provide a comma separated list of environment variables
+                    that should be forwarded to the remote. E.g. VAR1,VAR2,VAR3
+                    Vars prefixed with REACH are forwarded automatically
+
+Special commands:
+  auth              Authenticate with Reach cloud
+  config            Configure default Reach settings
+  local-install     Install Reach Local
+  local-up          Bring up Reach Local
+  local-down        Halt Reach Local
+  version           Display version
+|]
+
+data CliOptions = CliOptions
+  { cli_command :: (String, [String])
+  , cli_error :: Maybe String
+  , cli_forwardedEnvVars :: Maybe [String]
+  , cli_cloudOrLocal :: Maybe Bool -- True = cloud, False = local
+  , cli_connector :: Maybe String
+  , cli_project :: Maybe String
+  , cli_organization :: Maybe String
+  }
+
+emptyCliOptions :: CliOptions
+emptyCliOptions = CliOptions ("", []) Nothing Nothing Nothing Nothing Nothing Nothing
+
+parseCliOptions :: IO CliOptions
+parseCliOptions = flip go emptyCliOptions <$> getArgs
+ where
+  go args opts = case args of
+    ("-h":args')            -> go ("--help" : args') opts
+    ("-e":args')            -> go ("--env"  : args') opts
+    ("--help":_)            -> go ["local-help"] opts
+    ("--env":list:args')    -> go args' opts{ cli_forwardedEnvVars = Just $ splitOn "," list }
+    ("--cloud":args')       -> go args' opts{ cli_cloudOrLocal = Just True }
+    ("--local":args')       -> go args' opts{ cli_cloudOrLocal = Just False }
+    ("--connector":c:args') -> go args' opts{ cli_connector = Just c }
+    ("--project":p:args')   -> go args' opts{ cli_project = Just p }
+    ("--organization":o:args') -> go args' opts{ cli_organization = Just o }
```

Comment:
I think that `--org` should be a short-hand. But also, I think that when you specify a project like `@xbacked/staking`, then the `@xbacked` means the organization. I can see how there might be commands that need to know an org without a project, but I think that most of the time `--project` will be only thing used

Reply:
Thinking on it further, I'm not sure it even makes sense to have a project flag. Since a reach.toml file is required to recognize a project locally, it may as well just require the project be specified in it..

Reply:
I'll make these changes in the dedicated config pr